src/microbe_stage/Microbe.Contact.cs

```diff
@@ -509,11 +509,20 @@ public void Kill()
 
         var chunkScene = SpawnHelpers.LoadChunkScene();
 
+        // Local list for limiting number of organelles to spawn
+        var organellesAvailable = new List<PlacedOrganelle>(organelles);
+
         for (int i = 0; i < chunksToSpawn; ++i)
         {
             // Amount of compound in one chunk
             float amount = HexCount / Constants.CORPSE_CHUNK_AMOUNT_DIVISOR;
 
+            // If all possible organelles have been spawned or none available, end loop
+            if (organellesAvailable.Count <= 0)
+            {
+                break;
+            }
```

Comment:
This should NEVER happen because crunksToSpawn is divided by 3 on line 508; And if it DOES happen, this will cause the remaining compounds vanishing.